test/test_kaldi_compatibility.py

```diff
@@ -0,0 +1,53 @@
+"""Test suites for checking numerical compatibility against Kaldi"""
+import shutil
+import unittest
+import subprocess
+
+import kaldi_io
+import torch
+import torchaudio.functional as F
+
+
+def _exe_exists(cmd):
+    return shutil.which(cmd) is not None
+
+
+def _convert_args(**kwargs):
+    args = []
+    for key, value in kwargs.items():
+        key = '--' + key.replace('_', '-')
+        value = str(value).lower() if value in [True, False] else str(value)
+        args.append('%s=%s' % (key, value))
+    return args
+
+
+def _run_kaldi(command, input_tensor):
+    """Run provided Kaldi command, pass a tensor and get the resulting tensor
```

Comment:
Since kaldi is not installed in CI, the test is skipped in travis
```
(test/test_kaldi_compatibility.py::TestFunctional::test_sliding_window_cmn SKIPPED [100%]
```

For the other tests, we simply saved the output from kaldi, and compare against that. This allows us to run the test on CI. Let's do that.

If you want to keep the infrastructure you have added to run kaldi commands, we can do it too, along with the test just mentioned. We can rename this as e.g. `test_sliding_window_cmn_against_installed_kaldi`.

Reply:
My concern with the previous approach is that it adds overhead for both adding a new test and maintaining the test, which leads to the situation where every time a new Kaldi-compatible functionality is added, one of us has to add all the related files, because it is a complicated process to review.

When we run tests on Circle CI, we can bake Kaldi binaries into base test image. Making Kaldi accessible on CI should be easy.

Reply:
I do also prefer using kaldi commands to test against for new transforms, over testing against saved kaldi output files. This has the advantage of making sure we stay in sync with kaldi, and we can regenerate the exact command that produces the kaldi output we want.

Adding kaldi is easy (though there is always a risk that it won't run easily on some platform like windows), but we should isolate this change in a separate PR from the testing. And, since kaldi is not yet available in CircleCI, this test is not running and so is not showing us that `sliding_window_cmn` is compatible with kaldi.

Even if kaldi were available, we do not have convincing evidence in this PR's CircleCI that the new test is indeed providing the signal we want.

I see two options:
* We add kaldi in the testing build now.
* We add a second test against the kaldi output file, along with what you have here.

Let's go for the latter in this PR, and follow up with the former.